core/src/price_finding/optimization_price_finder.rs

```diff
@@ -193,28 +193,59 @@ pub struct OptimisationPriceFinder {
     io_methods: Arc<dyn Io + Send + Sync>,
     fee: Option<Fee>,
     solver_type: SolverType,
-    price_oracle: Box<dyn PriceEstimating + Sync>,
-    min_avg_fee_per_order: u128,
+    price_oracle: Box<dyn PriceEstimating + Send + Sync>,
+    gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
+    min_avg_fee_subsidy_factor: f64,
+    default_min_avg_fee_per_order: u128,
     internal_optimizer: InternalOptimizer,
 }
 
 impl OptimisationPriceFinder {
     pub fn new(
         fee: Option<Fee>,
         solver_type: SolverType,
-        price_oracle: impl PriceEstimating + Sync + 'static,
-        min_avg_fee_per_order: u128,
+        price_oracle: impl PriceEstimating + Send + Sync + 'static,
+        gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
+        min_avg_fee_subsidy_factor: f64,
+        default_min_avg_fee_per_order: u128,
         internal_optimizer: InternalOptimizer,
     ) -> Self {
         OptimisationPriceFinder {
             io_methods: Arc::new(DefaultIo),
             fee,
             solver_type,
             price_oracle: Box::new(price_oracle),
-            min_avg_fee_per_order,
+            gas_station,
+            min_avg_fee_subsidy_factor,
+            default_min_avg_fee_per_order,
             internal_optimizer,
         }
     }
+
+    async fn compute_min_average_fee_per_order(&self) -> u128 {
+        let eth_owl_xrate = match self.price_oracle.get_eth_price().await {
+            Some(owl_price) => (owl_price as f64) / 10f64.powi(18),
+            None => {
+                log::warn!("failed to find ETH price estimate");
+                return self.default_min_avg_fee_per_order;
+            }
+        };
+        let gas_estimate = match self.gas_station.estimate_gas_price().await {
+            Ok(gas) => pricegraph::num::u256_to_f64(gas.fast),
+            Err(err) => {
+                log::warn!("failed to estimate gas {:?}", err);
+                return self.default_min_avg_fee_per_order;
+            }
+        };
+
+        let subsidized_gas_price = (gas_estimate * eth_owl_xrate) / self.min_avg_fee_subsidy_factor;
+
+        const GAS_PER_ORDER: f64 = 120_000.0;
+        const FEE_ADJUSTMENT: f64 = 1_000.0;
```

Comment:
This is because of 1/1000, right? I was confused because the parameter is called "fee per order" so I thought it was already referring to the earned fee. It must mean something else, but then I'm also wondering if this takes into account that we only get half of the fee imbalance as a reward. Maybe @josojo can clarify.

Reply:
AFAIK fees get applied twice per trade (for buying and selling) of which half are burned. So we should collect 0.1% of each order.

Reply:
I'm also super confused why we multiply by a thousand. Shouldn't we compute `tx_cost_in_OWL = GAS_PER_ORDER * gas_price * eth_owl_xrate` which gives us some price in OWL and this is the minimum fee (maybe it has to be multiplied by 2 as the solver might consider overall fee and not just what the submitter gets credited)?

I would then in the end divide by the subsidy factor to make it very clear what the "real" economically viable value is and what the value is that we are willing to accept in the short term.

Reply:
This makes a lot of sense to me. I had taken the equation from @josojo's issue which also multiplied by 1000 which I assumed it was for fees.

Maybe @josojo or @marcovc can give some more details on the expected value for the `minAvgFeePerOrder` parameter.

Reply:
The computation was indeed incorrect. I have updated the test case to use @josojo's number from the issue and it seems to be OK now.